### ISSUE TITLE
Define product requirements for versioning frontend

### ISSUE THREAD

**Issue by neel-a-mehta** [seattle-uat/civiform#545]
Check out [these mocks](https://miro.com/app/board/o9J_lXULxwE=/?moveToWidget=3074457354952974840&cot=14) for how the versioning system should work; we haven't done much work on defining how the frontend would work.

Please create a product requirements doc in Google Docs that specifies the following, given the mockups and [other versioning issues](https://github.com/seattle-uat/civiform/labels/versioning) we've made:

- What each button on the mockups should do
- Changes you would recommend to the mocks for greater usability
- What counts as a "change"
- What updates count as important enough to merit an entry in the update history (does changing the order of questions count? do you have to add a brand new question for it to be important enough?)
- What you see when you hit "view" next to a date and time on the update portal (do you see just changes from the previous version? or a preview of the whole form as an applicant would see it? or a tree view showing small widgets for each question, as a sort of overview?)
- Do we need any other "states" for the form system besides draft, published, and obsolete? What does the "state machine" showing flows between each state look like?

**Comment by neel-a-mehta** [comment]
Here's a conversation we were having about what counts as an important-enough update to surface on the program list page:


> Does update history include changes to questions the program depends on? I.e. would changing a single question in a draft count as an "update" in the history for all the programs that use it?

> My initial thought is yes - any update that effected this program even if it wasn't authored in the context of this program. That said, it could get rather unwieldy, and not very meaningful if someone is updating widely used widgets/questions (like name address). IMO, it's unlikely those are receiving frequent update. 

> I would agree  -- while it may add clutter, it ensures that the admin doesn't miss anything. Better to over-communicate in this case. A compromise could be to show an icon next to each update showing how "drastic" it was. Edits to the questions that make up the program might get a low drastic-ness score, while adding new questions or deleting questions would get a higher score.

> I wonder if we can really provide assurance about how "drastic" a change might be.  I think I 
would keep it simple by showing that the program has an unpublished draft and the date/times of all the prior published versions
 
> ^^ Good point on the 'unpublished draft' showing here. I'll include that. RE above - I'm not sure that something being included in other forms or not is really correlated or an indication of if a change is more or less drastic a change is to an individual form, but maybe it's still informative?

**Comment by neel-a-mehta** [comment]
@HansEngebretsen @ndmckinley anything else you need clarified by product here?

**Comment by HansEngebretsen** [comment]
Is there a difference between a draft and previewing a version?  I'm assuming no - at least according to the mocks we did earlier her: https://miro.com/app/board/o9J_lXULxwE=/?moveToWidget=3074457354952974840&cot=14 , but if that's something we plan on incorporating, I'd add it.

**Comment by HansEngebretsen** [comment]
Also, this is maybe just related and not to be included here, but I would add a note about being able to preview a draft (since versions will be come drafts, and it's important to preview a version before re-deploying)

**Comment by neel-a-mehta** [comment]
> Is there a difference between a draft and previewing a version? I'm assuming no - at least according to the mocks we did earlier her: https://miro.com/app/board/o9J_lXULxwE=/?moveToWidget=3074457354952974840&cot=14 , but if that's something we plan on incorporating, I'd add it.

Depends. Suppose I'm working on a draft right now but I want to see if I could restore an old version instead. Could I click on that old version and see a summary / preview of it before I commit to it and throw out my current draft?

**Comment by neel-a-mehta** [comment]
Just met with Hans and Nathan about this; we've hashed out most of the product requirements, so it's mostly a UX question now: how to surface information most effectively to the user, and how to communicate complex things to them. Hans is going to take notes [on this Miro frame](https://miro.com/app/board/o9J_lXULxwE=/?moveToWidget=3074457354952974840&cot=14), so the task for this issue is to help him turn those notes into a new version of that Miro frame (i.e. make new mockups using the insights we just generated).
